int32 type
---
int32[] results
int32[] positions_z
int32[] positions_x